### ISSUE TITLE
[Bug]: Group with clipPath and children with shadow render error

### ISSUE THREAD

**Issue by byeval** [fabricjs/fabric.js#10282]
### CheckList

- [X] I agree to follow this project's Code of Conduct
- [X] I have read and followed the [Contributing Guide](https://github.com/fabricjs/fabric.js/blob/master/CONTRIBUTING.md)
- [X] I have read and followed the [Issue Tracker Guide](https://github.com/fabricjs/fabric.js/blob/master/CONTRIBUTING.md#%EF%B8%8F-issue-tracker)
- [X] I have searched and referenced existing issues and discussions
- [X] I am filing a **BUG** report.
- [X] I have managed to reproduce the bug **after** upgrading to the **latest** version
- [X] I have created an accurate and minimal reproduction

### Version

6.0.2

### In What environments are you experiencing the problem?

Chrome

### Node Version (if applicable)

None

### Link To Reproduction

https://codesandbox.io/p/sandbox/group-clippath-with-children-has-shadow-xplly3?file=%2Fsrc%2Findex.js%3A39%2C1&workspaceId=bc31a0d2-b26d-4ab3-b184-2d719079ef42

### Steps To Reproduce

1. create a rect with shadow as group background
2. group add a clipPath
3. render and get error

### Expected Behavior

render correctly without error

### Actual Behavior

render error and crashed
![image](https://github.com/user-attachments/assets/cd9511d9-ce94-452a-b60e-814b25aa24a2)

### Error Message & Stack Trace

```bash
TypeError
Cannot read properties of undefined (reading 'forEach')
Mr.createClipPathLayer
https://xplly3.csb.app/node_modules/fabric/dist/index.min.js:1:70862
Mr._drawClipPath
https://xplly3.csb.app/node_modules/fabric/dist/index.min.js:1:71135
Mr.drawObject
https://xplly3.csb.app/node_modules/fabric/dist/index.min.js:1:117783
Mr.render
https://xplly3.csb.app/node_modules/fabric/dist/index.min.js:1:69247
Mr.render
https://xplly3.csb.app/node_modules/fabric/dist/index.min.js:1:118096
Cn._renderObjects
https://xplly3.csb.app/node_modules/fabric/dist/index.min.js:1:32024
Cn.renderCanvas
https://xplly3.csb.app/node_modules/fabric/dist/index.min.js:1:31275
Cn.renderAll
https://xplly3.csb.app/node_modules/fabric/dist/index.min.js:1:133748
```

**Comment by asturur** [comment]
Thanks for the bug report.
Must have to be something related the recent clipPath changes.

**Comment by byeval** [comment]
Hi @asturur, thanks for your reply! I'm trying to read fabric source code to see if I can contribute too.

**Comment by byeval** [comment]
Sorry, I don't have enough time to address this issue,  I have read parts of the code but haven't found a solution yet.

**Comment by asturur** [comment]
Yes because if you comment willDrawShadow the group will be cached normally and everything works.
The shadow of a child wll break caching on purpose ( probably a dumb feature but it is what it is ).
This needs to be fixed

**Comment by byeval** [comment]
Thank you for your explanation. I will try to read more source code later.
